wp-content/themes/twentytwenty-child/functions.php

```diff
@@ -293,4 +338,232 @@ function update_students_settings() {
+    add_settings_field( 'show_address'   , 'Show address:'        , 'show_checkbox_address_ajax'  , 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_birthdate' , 'Show birthdate:'      , 'show_checkbox_birthdate_ajax', 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_class'     , 'Show class:'          , 'show_checkbox_class_ajax'    , 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_status'    , 'Show activity status:', 'show_checkbox_status_ajax'   , 'ajax-student-administration', 'students' );
+
+    register_setting( 'ajax-student-administration', 'show_country' );
+    register_setting( 'ajax-student-administration', 'show_address' );
+    register_setting( 'ajax-student-administration', 'show_birthdate' );
+    register_setting( 'ajax-student-administration', 'show_class' );
+    register_setting( 'ajax-student-administration', 'show_status' );
+}
+add_action( 'admin_init', 'ajax_register_settings' );
+
+/**
+ *  Callback for a custom setting
+ *  */
+function show_checkbox_country_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_country"] == 1 ) $checked = 'checked';
+    echo '<input type="checkbox" id="show_country" name="country_box_ajax" value="1" ' . $checked . '  />';
+}
+
+/**
+ *  Callback for a custom setting 
+ * */
+function show_checkbox_address_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_address"] == 1 ) $checked = 'checked';
+    echo '<input type="checkbox" id="show_address" name="address_box_ajax" value="1" ' . $checked . '  />';
+}
+
+/**
+ *  Callback for a custom setting 
+ * */
+function show_checkbox_birthdate_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_birthdate"] == 1 ) $checked = 'checked';
+    echo '<input type="checkbox" id="show_birthdate" name="birthdate_box_ajax" value="1" ' . $checked . '  />';
+}
+
+/**
+ *  Callback for a custom setting 
+ * */
+function show_checkbox_class_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_class"] == 1 ) $checked = 'checked';
+    echo '<input type="checkbox" id="show_class" name="class_box_ajax" value="1" ' . $checked . '  />';
+}
+
+/** 
+ * Callback for a custom setting 
+ * */
+function show_checkbox_status_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_status"] == 1 ) $checked = 'checked';
+    echo '<input type="checkbox" id="show_status" name="status_box_ajax" value="1" ' . $checked . '  /> </div> </form>';
+}
+
+/**
+ *  Function executed by AJAX when a checkbox is un/checked on the settings page 
+ * */
+function ajax_func() {
+    $options = get_option( 'show_ajax_settings' );
+    $options[ $_POST['option'] ] = 0;
+    if( $_POST['checked'] === 'true' ) {
+        $options[$_POST['option']] = 1;
```

Comment:
You missed this one.
```suggestion
        $options[ $_POST['option'] ] = 1;
```